daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/SchemaFileLocatable.scala

```diff
@@ -125,7 +125,21 @@ class XercesSchemaFileLocation(
 ) extends SchemaFileLocation(
     (if (xercesError.getLineNumber > 0) Some(xercesError.getLineNumber.toString) else None),
     (if (xercesError.getColumnNumber > 0) Some(xercesError.getColumnNumber.toString) else None),
-    schemaFileLocation.diagnosticFile,
+    ({
+      val sysId = xercesError.getSystemId
+      val diagnosticFile = schemaFileLocation.diagnosticFile
+      if (sysId == null) {
+        schemaFileLocation.diagnosticFile
+      } else {
+        val sysIdFile = new File(sysId)
+        val sysIdNormalizedPath = sysIdFile.getPath
+        // Return the diagnostic file if the ends of the paths match, otherwise return the new File
+        if (sysIdNormalizedPath.endsWith(diagnosticFile.getPath))
+          diagnosticFile
+        else
+          new File(sysIdNormalizedPath)
+      }
+    }),
```

Comment:
Can this change be reverted? Now that failures to load the validation schema result in an assertion, I think XercesSchemaFileLocation will only be created where schemaFileLocation.diagnosticFile is correct.